### PR TITLE
Started on an automated assembly step generator

### PR DESCRIPTION
**Pull request by jmwright**
This is an attempt to generate assembly step CAD models/assemblies using the device ID from devices.json. The code generates a placeholder device based on the dimensions in the database, generates the matching shelf, then attempts to put them together.

There is still missing data in the database to make this fully automated. For instance, where are the mounting hole locations in the hardware, if there are any? There needs to be some way to know if there are screws involved, and if so, where they go.

### CODE REVIEW

**Comment by julianstirling** [comment]
I know this is a draft, but I am going to push forward and merge as:

- This is clear progress
- Lots of things are half finished in master right now  
- I don't want to break everything when we do the split talked about in https://github.com/Wakoma/docs/issues/30

File: mechanical/components/cadquery/assembly_step_generator.py [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+    assy.add(shelf, name="shelf", color=cq.Color(0.996, 0.867, 0.0, 1.0))
+    assy.add(device, name="device", color=cq.Color(0.565, 0.698, 0.278, 1.0))
+
+    # Set up constraints that will keep the device in the correct relationship to the shelf
+    assy.constrain("shelf", "Fixed")
+    assy.constrain("device@faces@>X", "shelf@faces@<Y[-2]", "Axis")
+    assy.constrain("device@faces@<Z", "shelf@faces@<Z[-3]", "Plane")
+    assy.solve()
+
+    from cadquery.vis import show
```
Comment (julianstirling) [inline mechanical/components/cadquery/assembly_step_generator.py:40]: Following PEP8 guidance Pylint gets upset if imports are not at the top of the file.

https://realpython.com/absolute-vs-relative-python-imports/#styling-of-import-statements

I'm going to leave this as a comment. We'll know to fix it later as PyLint is complaining

**Review by julianstirling — APPROVED** [review]
This is certainly progress. I'm going to get this merged in before we split up the repos.

File: mechanical/components/cadquery/assembly_step_generator.py (hunk 1 above)
Comment (jmwright) [reply mechanical/components/cadquery/assembly_step_generator.py:40]: This is just temporary during model development.

File: mechanical/components/cadquery/assembly_step_generator.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+
+def main():
+    """
+    Allows a user to run this as a command line utility.
+    Needs to be run from the root of the repository for
+    the JSON file to be found.
+    """
+
+    import json
```
[the remark this answers is not in the document]
Comment (julianstirling) [reply mechanical/components/cadquery/assembly_step_generator.py:51]: Cheers. I tend to do what PyLint tells me unless I have a very good reason not to. I suppose the benefit of conventions is the code being structured in the way the python community expects.

**Pull request merged**